src/lib/runner.rs

```diff
@@ -354,6 +354,12 @@ pub(crate) fn run_task(flow_info: &FlowInfo, flow_state: &mut FlowState, step: &
             let mut updated_step = functions::run(&step);
             updated_step = environment::expand_env(&updated_step);
 
+            if updated_step.config.toolchain.is_none()
+                && flow_info.config.config.toolchain.is_some()
+            {
+                updated_step.config.toolchain = flow_info.config.config.toolchain.clone();
+            }
+
```

Comment:
if you look 2 lines up, you can see that i call external modules from this function to update the task.
so functions updates the task and than environment updates the task.
maybe we can expose in toolchain module, also a function to update the task.
this way, this function doesn't become too much of a bloat (it is already), and the logic is in the specific modules.
wdyt?

Reply:
that makes sense. I think this won't apply to all tasks yet, unfortunately, tasks that define a script are not wrapped with the toolchain change, right?

Reply:
oh... you raise a good point here. this whole PR might be a bit problematic as it will add the toolchain to all tasks including tasks that just do copy file.
all commands will become 'rustup sometoolchain original command'
so i'll get 'rustup stable cp file1 file2'

that is an issue.
so it might not be a practical feature.

maybe we want to limit it only to cargo commands? are we missing something in this case?
i don't care about scripts. scripts are there for a specific purpose, but if someone writes cargo test in a script instead of command and args, ya, they will loose capabilities (like they lose today that auto installation capability)

Reply:
> if someone writes cargo test in a script instead of command and args, ya, they will loose capabilities (like they lose today that auto installation capability)

this is exactly how I discovered it :rofl: